File: programmers/[programmers] 네트워크.py
def solution(n, computers):
    def dfs(num):
        for i in range(n):
            if visited[i]==1: continue
            if computers[num][i]==1:
                visited[i]=1
                dfs(i)

    answer = 0
    visited= [0]*n
    for i in range(n):
        if visited[i]==0:
            visited[i]=1
            dfs(i)
            answer+=1
    return answer

File: programmers/[programmers] 네트워크.js
function solution(n, computers) {
  const N = computers.length,
    M = computers[0].length;
  const visited = Array(N).fill(false);
  let count = 0;

  const DFS = (prevY) => {
    for (let i = 0; i < M; i++) {
      if (visited[i] === true) continue;
      if (computers[prevY][i] === 0 || prevY === i) continue;
      visited[i] = true;
      DFS(i);
    }
  };

  for (let i = 0; i < N; i++) {
    if (visited[i] === true) continue;
    count += 1;
    DFS(i);
  }
  return count;
}
